proto/regen/ecocredit/v1alpha2/tx.proto

```diff
@@ -475,6 +475,10 @@ message MsgBuy {
     // disable_partial_fill disables the default behavior of partially filling
     // buy orders if the requested quantity is not available.
     bool disable_partial_fill = 5;
+
+    // retirement_location is the optional retirement location for the credits
+    // which will be used only if disable_auto_retire is false.
+    string retirement_location = 6;
```

Comment:
we should probably add some validation check related to this location in `MsgBuy.ValidateBasic()`

Reply:
I think the `ValidateBasic` should be strictly minimal . User should pay for all possible operations, and this is only possible in `DeliverTx`. hence, it's better to have 2 methods:
* ValidateBasic
* Validate -- which is run at the beginning of Msg handler / RPC method, and can use state information

Reply:
Added a check to `ValidateBasic` that uses `ValidateLocation`. If a retirement location is provided, `ValidateBasic` will check if the string matches the location format.

I'm not sure I understand the purpose of the second method you are proposing @robert-zaremba. Why do we need to use state information to check the location again? The logic for setting the retirement amount and retirement location is already implemented. Is there another check that I am missing?

Reply:
See https://github.com/regen-network/regen-ledger/pull/670#discussion_r779768975